src/lib.rs

```diff
@@ -976,6 +986,17 @@ mod test {
             &[88, 183, 11, 7, 192, 224, 203, 107, 144, 162, 48, 78, 61, 223],
             &data[..]
         );
+        let mut cipher_data: Vec<Vec<u8>> = Vec::new();
+        cipher_data.push(data.to_vec());
+        for i in 0..100 {
+            data = "my secret data".to_owned().into_bytes();
+            let _ = cocoon.encrypt(&mut data).unwrap();
+            cipher_data.push(data.to_vec());
+            assert_ne!(
+                &cipher_data.last().unwrap(),
+                &cipher_data.get(i).unwrap()
+                )
+        }
```

Comment:
You can make it simpler:

```rust
let mut prev_data = vec![0];
for i in 1..100 {
    let data = vec![i];
    assert_ne!(&prev_data, &data);
    prev_data = data;
}
```

Reply:
AH I meant to have it check against all previous iterations to make sure the new one was distinct, not just the previous values. I.e. validate all forward encryptions don't match any previous. Might take a bit longer to run though.

Reply:
Made some changes